metaflow/extension_support.py

```diff
@@ -445,56 +587,80 @@ def _get_extension_packages():
-                                next_dirs.append(ext_list[-1])
-                    if next_dirs is not None:
-                        dirs[:] = next_dirs[:]
-
-        # Add all these new packages to the list of packages as well.
-        packages_to_add = list(packages_to_add)
-        packages_to_add.sort()
-        mf_pkg_list.extend(packages_to_add)
+            mf_pkg_list.append(package_name)
+            mf_ext_packages[package_name] = {
+                "root_path": None,
+                "meta_module": meta_module,
+                "files": files_to_include,
+            }
 
     # Sanity check that we only have one package per configuration file
     errors = []
     for m, packages in config_to_pkg.items():
         if len(packages) > 1:
             errors.append(
                 "\tPackages %s define the same configuration module '%s'"
-                % (", and ".join(packages), m)
+                % (", and ".join(["'%s'" % p for p in packages]), m)
+            )
+    for m, packages in meta_to_pkg.items():
+        if len(packages) > 1:
+            errors.append(
+                "\tPackages %s define the same meta module '%s'"
+                % (", and ".join(["'%s'" % p for p in packages]), m)
             )
     if errors:
         raise RuntimeError(
-            "Conflicts in %s configuration files:\n%s" % (EXT_PKG, "\n".join(errors))
+            "Conflicts in '%s' files:\n%s" % (EXT_PKG, "\n".join(errors))
         )
 
     extension_points_to_pkg.default_factory = None
     # Figure out the per extension point order
     for k, v in extension_points_to_pkg.items():
+        # v is a dict distributionName/packagePath -> (dict tl_name -> MFPackage)
         l = [v[pkg].values() for pkg in mf_pkg_list if pkg in v]
-        extension_points_to_pkg[k] = list(chain(*l))
-    return mf_pkg_list, extension_points_to_pkg
+        # In the case of the plugins.cards extension, we allow those packages
+        # to be ns packages so we only list the package once (in its first position).
+        # In all other cases, we error out if we don't have a configuration file for the
+        # package (either a __init__.py of an explicit mfextinit_*.py)
+        final_list = []
+        have_null_config = False
+        for pkg in chain(*l):
+            if pkg.config_module is None:
+                if k == "plugins.cards":
+                    # This is allowed here but we only keep one
+                    if have_null_config:
+                        continue
+                    have_null_config = True
+                else:
+                    package_path = package_name_to_path.get(pkg.package_name)
+                    if package_path:
+                        package_path = "at '%s'" % package_path
+                    else:
+                        package_path = "'%s'" % pkg.package_name
+                    raise RuntimeError(
+                        "Package %s does not define a configuration file for '%s'"
```

Comment:
An import on my local looks like the following 
<img width="518" alt="Screen Shot 2022-02-17 at 9 29 04 AM" src="https://user-images.githubusercontent.com/11920247/154537387-d95a4b59-d3f6-4a5f-b167-38ee56356ec4.png">

It ends up failing with this error. I added additional logs that print `k, pkg`
<img width="1165" alt="Screen Shot 2022-02-17 at 9 29 57 AM" src="https://user-images.githubusercontent.com/11920247/154537509-f34aebb1-d155-4bbf-a54d-95a660ba94ab.png">

Reply:
Good catch. I will fix this. I am trying to be more pedantic about well formed packages but missed the case of "only define a card and no plugins". I'll fix that. It should actually not even catch the `plugins` extension if there is no other file but I see now that for PYTHONPATH extensions, I don't do that properly. Ooh, this is going to be fun.